network/network-address/src/lib.rs

```diff
@@ -321,6 +321,45 @@ impl NetworkAddress {
             .push(Protocol::Handshake(handshake_version))
     }
 
+    /// Check that a `NetworkAddress` looks like a typical LibraNet address with
+    /// associated protocols.
+    pub fn is_libranet_addr(&self) -> bool {
+        use Protocol::*;
+
+        let suffix = if self.len() == 3 {
+            let (prefix, suffix) = self.as_slice().split_at(1);
+
+            match prefix {
+                [Memory(_)] => (),
+                _ => return false,
+            }
+
+            suffix
+        } else if self.len() == 4 {
+            let (prefix, suffix) = self.as_slice().split_at(2);
```

Comment:
so a canonical libra NetworkAddr looks like: prefix = ip4 or ip6 or dns + tcp and suffix = noiseIK or PeerIdExchange + Handshake

Two comments:

* why not just check that the first element is ip or dns, then that the second element is tcp, then that ... (so checking items one by one instead of two by two)
* looks like PeerIdExchange is also a test only thing right? So a `if cfg!(test)` flow might be better.

Reply:
hmm i think you might be right... can probably simplify this with cfg(test)...